layouts/layouts/src/main/java/com/microsoft/device/dualscreen/layouts/FoldableLayout.kt

```diff
@@ -90,18 +89,21 @@ open class FoldableLayout @JvmOverloads constructor(
     }
 
     private fun registerWindowInfoFlow() {
-        val activity = context as AppCompatActivity
-        val windowInfoRepository = activity.windowInfoRepository()
-        activity.lifecycleScope.launch(Dispatchers.Main) {
-            activity.lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
-                windowInfoRepository.windowLayoutInfo.collect { info ->
+        job = MainScope().launch {
+            WindowInfoTracker.getOrCreate(context)
+                .windowLayoutInfo(context as Activity)
+                .collect { info ->
                     viewModel.windowLayoutInfo = info
                     layoutController.foldingFeature = info.getFoldingFeature()
                 }
-            }
         }
     }
 
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        job?.cancel()
+    }
+
```

Comment:
If we use what we used before this should be go away

Reply:
Same explanation as the previous coroutine job comment.